### PR TITLE
moving implementation from ISS

### PR DESCRIPTION
**Pull request by semenodm**
TODO: add tests

### CODE REVIEW

File: src/main/scala/fs2/aws/sqs/SQSConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package fs2.aws.sqs
+
+import com.amazon.sqs.javamessaging.SQSConnection
+import javax.jms.MessageListener
+
+trait SQSConsumer[F[_]] {
+  def callback: MessageListener
+  def startConsumer()
+  def shutdown()
```
Comment (kiambogo) [inline src/main/scala/fs2/aws/sqs/SQSConsumer.scala]: explicit types here would be good

Comment (kiambogo) [inline src/main/scala/fs2/aws/sqs/SQSConsumer.scala]: should start and shutdown be in context of F?

File: src/main/scala/fs2/aws/sqs/SQSConsumerBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import javax.jms.{MessageListener, Session}
+
+import scala.language.higherKinds
+
+class SQSConsumerBuilder[F[_]](val sqsConfig: SqsConfig, val listener: MessageListener)(
+    implicit F: Effect[F])
+    extends ConsumerBuilder[F] {
+
+  val start: F[SQSConsumer[F]] = {
+    F.delay {
```
[the remark this answers is not in the document]
Comment (semenodm) [reply src/main/scala/fs2/aws/sqs/SQSConsumerBuilder.scala]: i think i need to remove context of F in SQSConsumer

File: src/main/scala/fs2/aws/sqs/SQSConsumer.scala (hunk 1 above)
Comment (semenodm) [reply src/main/scala/fs2/aws/sqs/SQSConsumer.scala]:
```def serve[A](stream: fs2.Stream[F, A])(implicit F: Effect[F]): fs2.Stream[F, A] = {
    fs2.Stream
      .bracket(start)(con => F.delay(con.shutdown()))
      .flatMap(con => fs2.Stream.eval(F.delay(con.startConsumer())).drain ++ stream)
  }``` both Shutdown and statConsumer are executed in F context

**Pull request merged**